apiserver/storage/storage.go

```diff
@@ -756,3 +756,82 @@ func (a *API) Destroy(args params.Entities) (params.ErrorResults, error) {
 	}
 	return params.ErrorResults{result}, nil
 }
+
+// Detach sets the specified storage attachments to Dying, unless they are
+// already Dying or Dead. Any associated, persistent storage will remain
+// alive.
+func (a *API) Detach(args params.StorageDetachParams) (params.ErrorResults, error) {
+	if err := a.checkCanWrite(); err != nil {
+		return params.ErrorResults{}, errors.Trace(err)
+	}
+
+	blockChecker := common.NewBlockChecker(a.storage)
+	if err := blockChecker.ChangeAllowed(); err != nil {
+		return params.ErrorResults{}, errors.Trace(err)
+	}
+
+	result := make([]params.ErrorResult, len(args.Detachments))
+	for i, arg := range args.Detachments {
+		storageTag, err := names.ParseTag(arg.StorageTag)
+		if err != nil {
+			result[i].Error = common.ServerError(err)
+			continue
+		}
+
+		var entityTag names.Tag
+		if arg.EntityTag != "" {
+			var err error
+			entityTag, err = names.ParseTag(arg.EntityTag)
+			if err != nil {
+				result[i].Error = common.ServerError(err)
+				continue
+			}
+		}
+
+		switch storageTag := storageTag.(type) {
+		case names.StorageTag:
```

Comment:
There's only one valid tag type - storage tag. They'd be less code if we just did a ParseStorageTag() directly above instead of ParseTag() and then checking the tag type here?